src/v/cloud_storage/remote_partition.cc

```diff
@@ -457,49 +457,19 @@ class partition_record_batch_reader_impl final
             query = model::offset_cast(config.start_offset);
         }
         // Find manifest that contains requested offset or timestamp
-        int retry_quota = 4;
-        while (retry_quota-- > 0) {
-            auto stm_start_offset
-              = _partition->_manifest_view->stm_manifest().get_start_offset();
-            if (!stm_start_offset.has_value()) {
-                vlog(
-                  _ctxlog.error, "STM manifest has no start offset; no data");
-                co_return;
-            }
-            auto cur = co_await _partition->_manifest_view->get_cursor(query);
-            if (cur.has_failure()) {
-                vlog(
-                  _ctxlog.error,
-                  "Failed to query spillover manifests: {}, query: {}",
-                  cur.error(),
-                  query);
-                co_return;
-            }
-            auto new_stm_start_offset
-              = _partition->_manifest_view->stm_manifest().get_start_offset();
-            if (likely(new_stm_start_offset == stm_start_offset.value())) {
-                _view_cursor = std::move(cur.value());
-                if (
-                  _view_cursor->manifest()->get().get_start_offset().value()
-                  == new_stm_start_offset) {
-                    _initial_stm_start_offset = new_stm_start_offset;
-                }
-                initialize_reader_state(
-                  _view_cursor->manifest().value(), config);
-                co_return;
-            }
+        auto cur = co_await _partition->_manifest_view->get_cursor(query);
+        if (cur.has_failure()) {
             vlog(
-              _ctxlog.info,
-              "STM manifest start offset moved from {} to {} due to spillover "
-              "or truncation; retrying lookup {}",
-              stm_start_offset,
-              new_stm_start_offset,
+              _ctxlog.error,
+              "Failed to query spillover manifests: {}, query: {}",
+              cur.error(),
               query);
+            co_return;
         }
-        vlog(
-          _ctxlog.warn,
-          "Couldn't initialize cursor for {} after retrying",
-          query);
+        _view_cursor = std::move(cur.value());
+        co_await _view_cursor->maybe_sync_manifest();
+        initialize_reader_state(_view_cursor->manifest().value(), config);
```

Comment:
If a scheduling point is inserted between L470 and L471, the `task_local_ptr` is invalidated and we'll throw an error that bubbles all the way up to the client. This is correct, but seems a bit expensive. Perhaps we can inline some retries here like we do in other places?

Reply:
Or we could use `with_manifest` here:
```suggestion
        co_await _view_cursor->with_manifest([this, config] (auto manifest) {
          initialize_reader_state(manifest, config);
        });
```

Reply:
Just did this here https://github.com/redpanda-data/redpanda/pull/12621

Actually the `task_local_ptr` doesn't get invalidated because it's created _after_ the sync.

Generally it seems like `_view_cursor->manifest().value()` as a pattern isn't helpful as far as task local protection is concerned.

Reply:
> Just did this here https://github.com/redpanda-data/redpanda/pull/12621
Actually the task_local_ptr doesn't get invalidated because it's created after the sync.
Generally it seems like _view_cursor->manifest().value() as a pattern isn't helpful as far as task local protection is concerned.

Is task local ptr solving some sort of problem that is unique and can't be solved in some other way? My understanding is that it is being used to address some lifetime issues. But, we haven't needed it yet. Why is now different?

Reply:
The underlying manifest held by `async_manifest_view` can be invalidated after any scheduling point. It's not possible to assert that the user is not doing something like
```
auto m = cursor.manifest();
co_await ...
m->...
```
And `task_local_ptr` solves it. Another way to do this is to remove the `async_manifest_view_cursor::manifest` method.